File: src/ascore/as/modules/EnumModule.java
package ascore.as.modules;

import ascore.as.modules.builtins.ModuleBuiltin;
import ascore.as.modules.core.ASModuleManager;
import ascore.as.modules.core.ASModuleFactory;

public enum EnumModule {
    builtins(ModuleBuiltin::charger),
    ;

    EnumModule(ASModuleFactory moduleFactory) {
        ASModuleManager.enregistrerModule(this, moduleFactory);
    }
}

File: src/ascore/as/modules/corekotlin/EnumModule.kt
package ascore.`as`.modules.corekotlin

import ascore.`as`.modules.EnumModule
import ascore.`as`.modules.core.ASModuleFactory
import ascore.`as`.modules.core.ASModuleManager

enum class EnumModule(moduleFactory: ASModuleFactory?) {
    ;

    init {
        ASModuleManager.enregistrerModule(EnumModule.valueOf(this.toString()), moduleFactory)
    }
}